sdk/examples/xo_python/sawtooth_xo/xo_client.py

```diff
@@ -140,13 +163,37 @@ def _send_xo_txn(self, name, action, space=""):
         )
 
         batch_list = self._create_batch_list([transaction])
-
-        result = self._send_request(
-            "batches", batch_list.SerializeToString(),
-            'application/octet-stream'
-        )
-
-        return result
+        batch_ids = [b.header_signature for b in batch_list.batches]
+
+        if wait and wait > 0:
+            wait_time = 0
+            start_time = time.time()
+            self._send_request(
+                "batches", batch_list.SerializeToString(),
+                'application/octet-stream'
+            )
+            while wait_time < wait:
+                statuses = self._get_statuses(batch_ids)
+                wait_time = time.time() - start_time
+
+                if all(s == 'COMMITTED' for s in statuses.values()):
+                    return 'Batches committed in {:.6} sec'.format(wait_time)
+                elif any(s == 'INVALID' for s in statuses.values()):
+                    return 'Error: Invalid batch'
+                elif any(s == 'UNKNOWN' for s in statuses.values()):
+                    return 'Error: Unknown batch'
```

Comment:
I would like these errors to be more informative, and `xo` specific. The concepts of _invalid_ or _unknown_ batches are possibly meaningless to the person using `xo`. I would probably say something along the lines of _"Something went wrong while creating your games. Try again later."_ Or whatever makes sense in an `xo` context.